src/control/server/storage_scm_prep_test.go

```diff
@@ -63,278 +57,12 @@ func (mp *mockPrepScm) PrepReset(ScmState) (bool, error) {
-	pmemOut := `{
-   "dev":"namespace%d.0",
-   "mode":"fsdax",
-   "map":"dev",
-   "size":"2964.94 GiB (3183.58 GB)",
-   "uuid":"842fc847-28e0-4bb6-8dfc-d24afdba1528",
-   "raw_uuid":"dedb4b28-dc4b-4ccd-b7d1-9bd475c91264",
-   "sector_size":512,
-   "blockdev":"pmem%d",
-   "numa_node":%d
-}
-`
-	onePmem, _ := parsePmemDevs(fmt.Sprintf(pmemOut, 1, 1, 0))
-	twoPmemsJson := "[" + fmt.Sprintf(pmemOut, 1, 1, 0) + "," + fmt.Sprintf(pmemOut, 2, 2, 1) + "]"
-	twoPmems, _ := parsePmemDevs(twoPmemsJson)
-
-	tests := []struct {
-		desc           string
-		errMsg         string
-		cmdOut         string
-		expPmemDevs    []pmemDev
-		expCommands    []string
-		lookPathErrMsg string
-	}{
-		{
-			desc:        "no namespaces",
-			cmdOut:      "",
-			expCommands: []string{cmdScmListNamespaces, cmdScmListNamespaces},
-			expPmemDevs: []pmemDev{},
-		},
-		{
-			desc:        "single pmem device",
-			cmdOut:      fmt.Sprintf(pmemOut, 1, 1, 0),
-			expCommands: []string{cmdScmListNamespaces, cmdScmListNamespaces},
-			expPmemDevs: onePmem,
-		},
-		{
-			desc:        "two pmem device",
-			cmdOut:      twoPmemsJson,
-			expCommands: []string{cmdScmListNamespaces, cmdScmListNamespaces},
-			expPmemDevs: twoPmems,
-		},
-		{
-			desc:           "ndctl not installed",
-			lookPathErrMsg: msgNdctlNotFound,
-			errMsg:         msgNdctlNotFound,
-		},
-	}
-
-	for _, tt := range tests {
-		t.Run(tt.desc, func(t *testing.T) {
-			log, buf := logging.NewTestLogger(t.Name())
-			defer ShowBufferOnFailure(t, buf)()
-
-			mockLookPath := func(string) (s string, err error) {
-				if tt.lookPathErrMsg != "" {
-					err = errors.New(tt.lookPathErrMsg)
-				}
-				return
-			}
-
-			mockRun := func(in string) (string, error) {
-				commands = append(commands, in)
-				return tt.cmdOut, nil
-			}
-
-			commands = nil // reset to initial values between tests
-
-			// not using mockPrepScm because we want to exercise
-			// prepScm
-			ss := newMockScmStorage(log, defaultMockExt(), nil, []DeviceDiscovery{MockModule()},
-				false, newPrepScm(log, mockRun, mockLookPath))
-
-			AssertEqual(t, ss.initialized, false, tt.desc+": expected SCM to start un-initialized")
-
-			if err := ss.Discover(); err != nil {
-				// if GetNamespaces fails, init should still be set
-				if tt.lookPathErrMsg != "" {
-					AssertEqual(t, ss.initialized, true, tt.desc+": expected SCM to be initialized even if ndctl is not installed")
-				} else {
-					t.Fatal(err)
-				}
-			}
-
-			pmemDevs, err := ss.prep.GetNamespaces()
-			if err != nil {
-				if tt.lookPathErrMsg != "" {
-					ExpectError(t, err, tt.lookPathErrMsg, tt.desc)
-					return
-				}
-				t.Fatal(tt.desc + ": GetNamespaces: " + err.Error())
-			}
-
-			AssertEqual(t, commands, tt.expCommands, tt.desc+": unexpected list of commands run")
-			AssertEqual(t, pmemDevs, tt.expPmemDevs, tt.desc+": unexpected list of pmem device file names")
-		})
-	}
-}
+// tests moved to storage/scm/ipmctl_test.go
```

Comment:
Should the mocks go with it? Are they still used?

Reply:
There are some layering issues that need to be resolved at some point. This PR starts that work, but I think we'll need to finish it with an effort on pulling apart the tests to make them closer to true unit tests.

Reply:
Sounds good.